xdsl/dialects/builtin.py

```diff
@@ -499,6 +499,8 @@ def __init__(
             value = IntAttr(value)
         if isinstance(value_type, int):
             value_type = IntegerType(value_type)
+        if isinstance(value_type, IntegerType) and value_type.bitwidth == 1:
+            value = IntAttr(-abs(value.data))
```

Comment:
I don't understand this special case.
Why do we accept `1` as input for `i1` values?
I feel we should rather have a boolean constructor here that works for `IntegerType(1)`, rather
than doing this sneaky conversion?

Reply:
So it's an incremental step towards normalizing in the general signless case. I guess actually here I don't check for it being signless, which is a problem. Actually I can probably revert this chunk and the rest of the changes should still hold.